Automattic/wp-calypso
# Site Settings: Update the General Settings to use the redux actions

This follows #9056 

Here, I drop the usage of `formGeneral` mixin which used the legacy `saveSettings`, `fetchSettings` methods to retrieve and update Site Settings. Instead, we use `QuerySiteSettings`   and redux actions.

I also added a new HoC component `formFields` that mimics the dirtyLinkState mixin to handle form field updates and dirty fields without relying on the deprecated `linkState` react mixin. (no more warnings)

once #9088 merged, we could switch this form to an ES6 class.

 **Testing instructions**

 * Navigate the general settings page
 * Try to update all the different settings, make sure everything is working properly

**Questions**

 - What do you think of the HoC usage to replace our old mixins? Do you think of a better alternative?
 - Same question as #9088 Where do we put these HoC

cc @aduth
